pkg/capacity/manager.go

```diff
@@ -144,34 +144,46 @@ func getOptimalTargetClusters(preferredCluster string, getMemberClusters cluster
 	if len(members) == 0 {
 		return emptyTargetCluster
 	}
+	var filteredMembers []*cluster.CachedToolchainCluster
+	for _, member := range members {
+		// if cluster-role labels were provided, it will check for matching on the member labels
+		// if no clusterRoles labels are required, then the function will return all member cluster with the `tenant` cluster role label
+		if hasClusterRoles(clusterRoles, member) {
+			filteredMembers = append(filteredMembers, member)
+		}
+	}
+
+	// return empty string if no members available with roles
+	if len(filteredMembers) == 0 {
+		return emptyTargetCluster
+	}
 
 	// if the preferred cluster is provided and it is also one of the available clusters, then the same name is returned, otherwise, it returns the first available one
 	if preferredCluster != "" {
-		for _, m := range members {
+		for _, m := range filteredMembers {
 			if preferredCluster == m.Name {
 				return []string{m.Name}
 			}
 		}
 	}
 
+	// extract only names of the available clusters
 	var memberNames []string
-	for _, member := range members {
-		// if cluster-role labels were provided, it will check for matching on the member labels
-		// if no clusterRoles labels are required, then the function will simple return true
-		if hasClusterRoles(clusterRoles, member) {
-			memberNames = append(memberNames, member.Name)
-		}
-	}
-	// return empty string if no members available with roles
-	if len(memberNames) == 0 {
-		return emptyTargetCluster
+	for _, m := range filteredMembers {
+		memberNames = append(memberNames, m.Name)
 	}
```

Comment:
just a side note: you could have extracted just the names while filtering them based on the roles - you don't need the whole `CachedToolchainCluster` types, only the names. But it's just a detail.

Reply:
Good point! Not sure why I've created two lists in the first place 🤦 

Fixed in [f16977a](https://github.com/codeready-toolchain/host-operator/pull/736/commits/f16977a51010e145b15d0052602eccbe8879a9fc) 

Thanks!